src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoNetworkTask.java

```diff
@@ -111,7 +111,8 @@ private void routeItems(Location inputNode, Block inputTarget, int frequency, Ma
                 if (inv.getItem(previousSlot) == null) {
                     inv.setItem(previousSlot, stack);
                 }
-                else {
+                // Making sure the item cannot be added to another slot before dropping it on ground
+                else if((stack = inv.addItem(stack).get(0)) != null){
```

Comment:
```suggestion
                else if (!inv.addItem(stack).isEmpty()) {
```

Reply:
> 

This will lead to duplication. Let's say some of the items were deposited into the source inventory, you are ignoring the leftovers and you deposit the old item, which will result in a possible duplication.

Reply:
Oh you are right. Yeah, in that case your solution would be better.
I think I really need a break or some sleep o_O